research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,351 @@
+This basically corresponds to the [Dolev-Yao model](https://en.wikipedia.org/wiki/Dolev%E2%80%93Yao_model).
+
+An entity with this power would, in practice, also have the power of the internal linearly scaling nodes attacker.
+
+## Attack-based Threat Analysis
+
+The following lists various attacks including the weakest attacker model in which the attack can be successfully performed.
+The respective attack can be performed in all stronger attacker models as well.
+
+An attack is considered more powerful if it can be successfully performed in a weaker attacker model.
+
+If not stated otherwise, we look at these attacks with respect to their capability to deanonymize the message sender.
+
+### Scope
+
+In this post, we introduce a simple tightly scoped threat model for Waku v2 Relay, which will be extended in the course of this article series.
+
+In this first post, we will look at the relay protocol in isolation.
+Even though many threats arise from layers Waku relay is based on, and layers that in turn live on top of relay,
+we want to first look at relay in isolation because it is at the core of Waku v2.
+Addressing and trying to solve all security issues of a complex system at once is an overwhelming task, which is why we focus on the soundness of relay first.
+
+This also goes well with the modular design philosophy of Waku v2, as layers of varying levels of security guarantees can be built on top of relay, all of which can relay on the guarantees that Waku provides.
+Instead of looking at a multiplicative explosion of possible interactions, we look at the core in this article, and cover the most relevant combinations in future posts.
+
+Further restricting the scope, we will look at the data field of a relay message as a black box.
+In a second article on Waku v2 relay, we will look into the data field, which according to the [specification of Waku v2 relay](https://rfc.vac.dev/spec/11/#message-fields) must be a [Waku v2 message](https://rfc.vac.dev/spec/14/).
+We only consider messages with version field `2`, which indicates that the payload has to be encoded using [35/WAKU2-NOISE](https://rfc.vac.dev/spec/35/).
+
+### Prerequisite: Get a Specific Position in the Network
+
+Some attacks require the attacker node(s) to be in a specific position in the network.
+In most cases, this corresponds to trying to get into the mesh peer list for the desired pubsub topic of the victim node.
+
+In libp2p gossipsub, and by extension Waku v2 relay, nodes can simply send a graft message for the desired topic to the victim node.
+If the victim node still has open slots, the attacker gets the desired position.
+This only requires the attacker to know the gossipsub multiaddress of the victim node.
+
+A linearly scaling nodes attacker can leverage DHT based discovery systems to boost the probability of evil nodes being returned, which in turn significantly increases the probability of attacker nodes ending up in the peer lists of victim nodes.
```

Comment:
evil -> bad/misbehaving?

Reply:
We used "evil" as short hand for "node controlled by the attacker".
Bad/Misbehaving could be unintentional or not coordinated by the attacker.
Ill use the more formal "malicious". Wdyt? see: 0c514c45c